src/Transition.js

```diff
@@ -166,6 +167,9 @@ class Transition extends React.Component {
   // }
 
   componentDidMount() {
+    const node = ReactDOM.findDOMNode(this)
+    // This is necessary to make a transition happen
+    reflow(node)
```

Comment:
I thought we didn't want to unconditionally do this for all transitions? cc @jquense

Reply:
we don't i don't think

Reply:
That makes sense so I've updated this PR to trigger reflow only that the component has an appearing animation.